File: DelegationDesignPattern/src/main/java/org/example/example01v1/java/Room.java
package org.example.example01v1.java;

public class Room {
    private final WindowActionListener window=new Window();
    private final DoorActionListener door=new Door();
    public void closeDoor(){
        door.closeDoor();
    }
    public void openDoor(){
        door.openDoor();
    }
    public void closeWindow(){
      window.closeWindow();
    }
    public void openWindow(){
        window.openWindow();
    }

}

File: DelegationDesignPattern/src/main/java/org/example/example01v1/kotlin/Room.kt
package org.example.example01v1.kotlin

import org.example.example01v1.java.WindowActionListener

//class Room:WindowListener by Window(),DoorActionListener by Door()
class Room:WindowActionListener,DoorActionListener {
    private val window=Window()
    private val door=Door()
    override fun openWindow() {
       window.openWindow()
    }

    override fun closeWindow() {
      window.closeWindow()
    }

    override fun openDoor() {
        door.openDoor()
    }

    override fun closeDoor() {
        door.closeDoor()
    }

}